src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -105,6 +105,14 @@ function checkAndWarnForMutatedStyle(style1, style2, component) {
   if (shallowEqual(style1, style2)) {
     return;
   }
+  // Don't show mutated style warning if a value is NaN since the styles weren't mutated.
+  for (var key in style1) {
+    var value = style1[key];
+
+    if (isNaN(value) && ('' + value) === 'NaN') {
```

Comment:
Thanks for that. That's basically what I was trying to emulate but somehow missed the more specific Number.isNaN in my search. Updated with this change.